pages/profile/[username].tsx

```diff
@@ -0,0 +1,165 @@
+                    </LoaderWrapper>
+                </div>
+                {/* Input Message */}
+                {!isLoading && <div className={data?.allow_anon_dm ? "w-full bg-white md:max-w-M lg:max-w-M xl:max-w-M p-2 fixed bottom-0 flex flex-row gap-[6px] z-[9999]" : "max-w-[375px] p-2 mb-4 bg-gray05 fixed bottom-0 flex flex-row gap-[6px] rounded-lg z-[9999]"}>
+                    <Image className="rounded-full" src='/image/anonIcon.svg' alt="anon icon" width={24} height={24} />
+                    {data?.allow_anon_dm ? (
+                        <>
+                            <div className="flex flex-grow items-ce bg-gray05 rounded-xl py-1 px-2">
+                                <textarea ref={textAreaRef} rows={1} onChange={(e) => {
+                                    setMessage(e.target.value)
+                                }} className="bg-transparent min-h-[24px] w-full"
+                                    style={{ resize: 'none' }}
+                                    placeholder="Send a message..." />
+                                <Toggle
+                                    checked={true}
+                                    className="bg-white"
+                                    defaultChecked={true}
+                                    onClick={() => showDownloadToast()}
+                                    icons={{
+                                        checked: <text className="text-xs font-normal text-cyan leading-[18px]">On</text>,
+                                        unchecked: null,
+                                    }}
+                                />
+                            </div>
+                            <button className="rounded-full flex-shrink-0 bg-cyan h-8 w-8 flex items-center justify-center" onClick={() => {
+                                localStorage.setItem(MessageEnum.tempMessage, message);
+                                router.push('/verification')
+                            }}>
+                                <Image src='/image/sendMessageicon.svg' alt="icon send" width={32} height={32} />
+                            </button>
+                        </>
+                    ) : (
+                        <>
+                            <div className="flex flex-grow bg-gray05 rounded-xl py-1 px-2">
+                                <div className="min-h-[24px] w-[230px] text-sm">
+                                    This user does not want to receive an anonymous messages.
+                                </div>
+                            </div>
+                        </>
+                    )}
+                </div>}
+            </LayoutContainer>
+        </BaseContainer>
+    )
+}
+
+export const getServerSideProps = (context: GetServerSidePropsContext) => {
+    const { username } = context.query; // Retrieve the URL parameter from context.query
+
+    return {
+        props: {
+            username,
+        },
+    };
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

It's good to validate the `username` parameter before returning it as a prop. This can help prevent potential security issues related to unvalidated user input.

```diff
- const { username } = context.query; // Retrieve the URL parameter from context.query
+ const { username } = context.query;
+ if (typeof username !== 'string' || !username.trim()) {
+     return {
+         notFound: true,
+     };
+ }

- return {
-     props: {
-         username,
-     },
- };
+ return {
+     props: {
+         username,
+     },
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->